### PR TITLE
Docs carto namespace

### PR DESCRIPTION
**Pull request by IagoLast**
- Document carto namespace
- Improve carto.source.Dataset docs
- Improve carto.source.SQL docs

### CODE REVIEW

File: src/api/source/dataset.js [hunk 1]
```diff
@@ -6,7 +6,19 @@ import CartoValidationError from '../error-handling/carto-validation-error';
 export default class Dataset extends BaseWindshaft {
 
     /**
-     * Create a carto.source.Dataset.
+     * A dataset defines the data that will be displayed in a layer and is equivalent
+     * to a table in the server.
+     * 
+     * If you have a table named `european_cities` in your CARTO account you could load all the
+     * data in a layer using a `carto.Dataset`.
```
Comment (Jesus89) [inline src/api/source/dataset.js]: carto.source.Dataset

File: src/api/style.js [hunk 2]
```diff
@@ -295,11 +295,11 @@ export default class Style {
...
-         * @property {carto.style.expression.Base} color
-         * @property {carto.style.expression.Base} width
-         * @property {carto.style.expression.Base} strokeColor
-         * @property {carto.style.expression.Base} strokeWidth
-         * @property {carto.style.expression.Base} order
+         * @property {carto.style.expressions.Expression} color
+         * @property {carto.style.expressions.Expression} width
+         * @property {carto.style.expressions.Expression} strokeColor
+         * @property {carto.style.expressions.Expression} strokeWidth
+         * @property {carto.style.expressions.Expression} order
```
Comment (Jesus89) [inline src/api/style.js]: This is redundant. We do not have carto.source.Source for the Base class, we use carto.source.Base instead. IMO we should do the same here with the expressions.

Comment (IagoLast) [reply src/api/style.js]: I think this is out of the scope of this PR, just changed an inexistent documented object for the real one ;) 

In any case Im totally against emulating java in javascript including fake abstract "clasess". We can discuss this in the future

File: src/core/style/expressions/expression.js [hunk 3]
```diff
@@ -3,7 +3,11 @@ import { blend, animate } from '../functions';
 import * as schema from '../../schema';
 
 /**
- * @description Abstract expression class
+ * Abstract expression class
+ * All expressions listed in  {@link carto.style.expressions} inherit from this class so any of them
+ * they can be used where an Expression is required as long as the types match. 
+ * 
+ * This means that you can't a numeric expression where a color expression is expected.
```
Comment (davidmanzanares) [inline src/core/style/expressions/expression.js]: you can't  **use**  a numeric expression where a color expression is expected.

File: src/api/style.js (hunk 2 above)
Comment (davidmanzanares) [reply src/api/style.js]: I think the problem is just the naming of the `Expression` class, in any case, it is outside the PR scope

**Pull request merged**